File: Python/Functional-2/no_teen.py

# given a list of integers, return a list of those numbers, omitting any that are between 13 and 19 inclusive
def no_teen(nums):
    new_list = []

    for x in nums:
        if x < 13 or x > 19:
            new_list.append(x)

    return new_list

print(no_teen([12, 13, 19, 20]))
print(no_teen([1, 14, 1]))
print(no_teen([15]))

File: JavaScript/Functional-2/noTeen.js

// given a list of integers, return a list of those numbers, omitting any that are between 13 and 19 inclusive
const noTeen = (nums) => {
    const newList = nums.filter(n => n < 13 || n > 19);

    return newList;
};

console.log(noTeen([12, 13, 19, 20]));
console.log(noTeen([1, 14, 1]));
console.log(noTeen([15]));
